src/main/java/backend/Logic.java

```diff
@@ -208,6 +208,11 @@ public String getDefaultRepo() {
         return models.getDefaultRepo();
     }
 
+    public void removeRepository(String repoId) {
+        repoIO.removeRepository(repoId);
+        models.removeRepoModelById(repoId);
```

Comment:
@dariusf How do I update `models`? Here it doesn't seem that the changes is applied to the `models`.